src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -60,27 +73,36 @@
   private final int _latencyPercentileGranularityMs;
   private final AtomicBoolean _running;
   private final int _latencySlaMs;
+  private final String _zkConnect;
+  private final String _topic;
+  private final int _intervalPartitionLeaderExecutor;
+  private final ScheduledExecutorService _handlePartitionLeaderInfoExecutor;
+  private final Map<Integer, Broker> _partitionToBroker;
 
   public ConsumeService(Map<String, Object> props, String name) throws Exception {
     _name = name;
     Map consumerPropsOverride = props.containsKey(ConsumeServiceConfig.CONSUMER_PROPS_CONFIG)
       ? (Map) props.get(ConsumeServiceConfig.CONSUMER_PROPS_CONFIG) : new HashMap<>();
     ConsumeServiceConfig config = new ConsumeServiceConfig(props);
-    String topic = config.getString(ConsumeServiceConfig.TOPIC_CONFIG);
-    String zkConnect = config.getString(ConsumeServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
+    _topic = config.getString(ConsumeServiceConfig.TOPIC_CONFIG);
+    _zkConnect = config.getString(ConsumeServiceConfig.ZOOKEEPER_CONNECT_CONFIG);
     String brokerList = config.getString(ConsumeServiceConfig.BOOTSTRAP_SERVERS_CONFIG);
     String consumerClassName = config.getString(ConsumeServiceConfig.CONSUMER_CLASS_CONFIG);
     _latencySlaMs = config.getInt(ConsumeServiceConfig.LATENCY_SLA_MS_CONFIG);
     _latencyPercentileMaxMs = config.getInt(ConsumeServiceConfig.LATENCY_PERCENTILE_MAX_MS_CONFIG);
     _latencyPercentileGranularityMs = config.getInt(ConsumeServiceConfig.LATENCY_PERCENTILE_GRANULARITY_MS_CONFIG);
+    _intervalPartitionLeaderExecutor = config.getInt(ConsumeServiceConfig.PARTITION_TO_LEADER_REFRESH_INTERVAL_MS_CONFIG);
     _running = new AtomicBoolean(false);
+    _partitionToBroker = new ConcurrentHashMap<>();
 
     for (String property: NONOVERRIDABLE_PROPERTIES) {
       if (consumerPropsOverride.containsKey(property)) {
         throw new ConfigException("Override must not contain " + property + " config.");
       }
     }
 
+    _handlePartitionLeaderInfoExecutor = Executors.newSingleThreadScheduledExecutor(new HandlePartitionLeaderInfoThreadFactory());
```

Comment:
You need to shutdown that executor in the `stop()` method

Reply:
Ohh good catch..